src/components/Participants/Participants.js

```diff
@@ -5,21 +5,101 @@
  * of the MIT license.  See the LICENSE.txt file for details.
  */
 
-import React from 'react';
+import React, { useEffect } from 'react';
 import { useSelector } from 'react-redux';
 
 import Participant from '../Participant';
 
+/*
+ * See setParticipantVars
+ */
+function getCssConfig(element, property, def) {
+  let value = element.style.getPropertyValue(property);
+
+  if (value === "") {
+    return def;
+  } else {
+    return Number(value);
+  }
+}
+
+/*
+ * Calculates the optimal width for the participants inside the div and stores
+ * the calculated values (size and number of squares) as CSS variables in the
+ * given div.
+ *
+ * The calculation can be configured with some CSS variables in the div.
+ *
+ * The range of possible values for --partSlotWidth can be configured with
+ * --partMaxSlotWidth, --partMinSlotWidth and --partStepSlotWidth.
+ *
+ * By default, the algorithm assumes a square representation for each
+ * participant. That can be configured with the variables --partHeightFactor,
+ * --partHeightExtra and --partWidthExtra.
+ */
+function setParticipantVars(div) {
+  let totalWidth = div.offsetWidth;
+  let totalHeight = div.offsetHeight;
+  let qty = div.querySelectorAll(".participant").length;
+
+  let max = getCssConfig(div, "--partMaxSlotWidth", 160);
+  let min = getCssConfig(div, "--partMinSlotWidth", 80);
+  let step = getCssConfig(div, "--partStepSlotWidth", 4);
+
+  let heightFactor = getCssConfig(div, "--partHeightFactor", 1);
+  let heightExtra = getCssConfig(div, "--partHeightExtra", 0);
+  let widthExtra = getCssConfig(div, "--partWidthExtra", 0);
```

Comment:
NP: it's a common practice to use `const` instead of `let` when you're not going to reassign the variable.

Reply:
Done (I hope).